### ISSUE TITLE
Anomaly "Not an arity" on typing sort-polymorphic inductive applied to non-substituted "Meta"

### ISSUE THREAD

**Issue by herbelin** [coq/coq#7645]
#### Version

At least from 8.4 to 8.8 (and an error in 8.3).

#### Description of the problem

```
Inductive I (A : Type) :=.
Axiom r : forall (A:Type) (P:A -> Type) (t : A), P t = P t.
Goal True.
rewrite r with (P:=I).
(* Anomaly "Not an arity." *)
```
This is caused by `I` being applied to a `Meta` of type another (defined) `Meta`. A possible fix would be that `EConstr.to_constr` instantiates not only defined `Evar` but also defined `Meta`.

**Comment by Zimmi48** [comment]
This doesn't raise an anomaly anymore but "Found no subterm matching "I ?M162" in the current goal.".
Can you confirm that this mean that the issue has been fixed?

**Comment by herbelin** [comment]
> This doesn't raise an anomaly anymore but "Found no subterm matching "I ?M162" in the current goal.".
> Can you confirm that this mean that the issue has been fixed?

Suspectingly fixed by #13108 indeed.
